.github/workflows/mavenBuild.yml

```
@@ -88,16 +92,14 @@ jobs:
         target: .github/matcher
     - run: echo "::add-matcher::.github/matcher/${{ steps.api-tools-matcher.outputs.filename }}"
     - name: Build with Maven
-      uses: coactions/setup-xvfb@6b00cf1889f4e1d5a48635647013c0508128ee1a
-      with:
-       run: >- 
-        mvn --batch-mode -V -U -e
+      run: >-
+        ${{ runner.os == 'Linux' && 'xvfb-run' || '' }}
+        mvn --batch-mode --no-transfer-progress --show-version --update-snapshots --errors
         ${{ inputs.without-EF-infra && '-Dtycho.version=4.0.12' || '' }}
-        -ntp
         -Dcompare-version-with-baselines.skip=false
         -Pbree-libs
         -Papi-check
-        --fail-at-end
+        --fail-at-end -Dmaven.test.failure.ignore=true
```

I don't think such generic properties should be set in a shared action. If this is wanted (and projects might decide differently) its better to either pass as an extra option in the calling workflow or put into `maven.config`.

---

It's a generic option yes, but I assume that each project actually would want that or at least would not object.
What do you think speaks against this?

---

> What do you think speaks against this?

1. It is bad practice , as it for example will build downstream projects that then fail with more tests because they rely on correct behavior. And still the "bad" module will be marked as `SUCCESS`.
2. "what projects actually want" is changing / different and you can't override the option if given on the commandline
3. I even remember we have used that option in the past and the people complained that the check shows green even it was failing tests
4. If the additional check fails to run (for whatever reason) then you even won't get a feedback at all

Actually we should even think about removing fail at end... If something is wrong we should report a failure as soon as possible and not waste resources for things that *might* still work.

---

I have never though about this option, but when doing so now, I also see some benefits of it:
- People would be better notified about any kind of failure not related to tests. I have seen merges despite different kinds of failures as they were "hidden" by the fact that there was a test failure which was just considered to be the only cause for the workflow to fail.
- The results would be obviously "cleaner" as a test failure only makes one check fail, so it removes the manual effort take a look at multiple check results in case of a failure.

But I also see the drawback that "success" for an execution with test failure is a bit strange and probably there should an option for consuming projects to opt-out.

> 4\. If the additional check fails to run (for whatever reason) then you even won't get a feedback at all

This is quite important point. Even if the additional check does not fail, it takes some time (seconds) until its done and shows up. That check is not shown as "pending" but just appears several seconds after the builds finished. I have seen people waiting actively for the builds to finish to directly merge the change, so this could result in PRs being merged right after all checks become green even though few seconds later a failure pops up.
Thus, I think we should somehow ensure that either the test results check shows up as "pending" or that the builds wait for the test results to show up before finishing (but then a build would only be marked as completed when all of them are).

> Actually we should even think about removing fail at end... If something is wrong we should report a failure as soon as possible and not waste resources for things that already are proven to fail.

This is something we cannot do as we have sporadically failing tests and people are (unfortunately) used to merging PRs with such a failing referring to the issue reporting that failure. So this could easily lead to test failure that would occur after such a known one to stay undiscovered.

---

> I even remember we have used that option in the past and the people complained that the check shows green even it was failing tests

I would say that each failing check should be inspected/understood in detail.

> * The results would be obviously "cleaner" as a test failure only makes one check fail, so it removes the manual effort take a look at multiple check results in case of a failure.

That's my main motivation as I find it very annoying to have to check two logs for one failure, just to be sure nothing is hidden. Of course we could also 'just' fix all flaky tests.

> * People would be better notified about any kind of failure not related to tests. I have seen merges despite different kinds of failures as they were "hidden" by the fact that there was a test failure which was just considered to be the only cause for the workflow to fail.

There can be actually two kinds of hidden. Hidden in the sense that the committer does not inspect the build-log because it is assumed that the test-failure is unrelated. And hidden that a failing test-project prevents the execution of another one that also has test-failures that don't occur because they are not executed. Although this is currently unlikely as we run with `-fae` and test inter-dependencies are at least very uncommon. But that's basically Heiko's last point.